### PR TITLE
HTML-634 Added tests for fail checking

### PR DESCRIPTION
**Pull request by CBft9MFGIb0Xd**
https://issues.openmrs.org/browse/HTML-634

### CODE REVIEW

File: api-tests/src/test/java/org/openmrs/module/htmlformentry/RegressionTest.java [hunk 1]
```diff
@@ -133,6 +133,33 @@ public void testResults(SubmissionResults results) {
...
+				return new String[] { "Date:", "Location:", "Provider:", "Weight:" };
+			}
+
+			@Override
+			public void setupRequest(MockHttpServletRequest request, Map<String, String> widgets) {
+			}
+
+			@Override
+			public void testResults(SubmissionResults results) {
+				results.assertErrors(3); // date, location, and provider, weight are required
```
Comment (mogoodrich) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/RegressionTest.java]: I believe that the comment is wrong.  For this test (and all 4 of the form failure tests that you added) the only requirements are that there is a date, location, and provider.  (That is why each one only has 3 errors).

File: api/src/test/java/org/openmrs/module/htmlformentry/widget/CheckboxWidgetTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CheckboxWidgetTest {
+
+    @Test
+    public void testHidden() throws Exception {
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/htmlformentry/widget/CheckboxWidgetTest.java]: This should be called something like "generateHtml_shouldGenerateProperHtml".

**Comment by mogoodrich** [comment]
Please address the comments and assign yourself to the ticket HTML-634 (I've marked it ready for work, so you should be able to do so now).

Thanks!

File: api-tests/src/test/java/org/openmrs/module/htmlformentry/RegressionTest.java (hunk 1 above)
Comment (CBft9MFGIb0Xd) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/RegressionTest.java]: Are the new four tests unnecessary then?

**Comment by dkayiwa** [comment]
Closing this because it does not do as advertised in the commit message.

**Pull request closed without merging**